.github/workflows/integration.yml

```diff
@@ -0,0 +1,321 @@
+name: Integration Suite
+
+on: ['push']
+
+env:
+  IMAGE_NAME: learninglocker/xapi-service
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    env:
+      NPM_CONFIG_LOGLEVEL: warn
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+          cache: 'yarn'
+
+      - name: Installing Dependencies
+        run: yarn install --ignore-engines --frozen-lockfile
+
+      - name: Compiling Code
+        run: yarn build
+
+      - name: Linting Code
+        run: yarn lint-ci
+
+  test-local:
+    name: Test Local
+    needs: build
+    runs-on: ubuntu-latest
+    env:
+      WINSTON_CONSOLE_LEVEL: none
+      EXPRESS_PORT: 1337
+
+    services:
+      mongodb:
+        image: mongo:3.7.9
+        env:
+          MONGO_INITDB_DATABASE: test
+          ALLOW_EMPTY_PASSWORD: yes
+        ports:
+          - 27017:27017
+        options: --health-cmd="mongo" --health-interval=10s --health-timeout=5s --health-retries=5
+
+      redis:
+        image: redis:3.2.12
+        env:
+          REDIS_DATABASE: test
+          ALLOW_EMPTY_PASSWORD: yes
+        ports:
+          - 6379:6379
+        options: --health-cmd="redis-cli ping"
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+          cache: 'yarn'
+
+      - name: Installing Dependencies
+        run: yarn install --ignore-engines --frozen-lockfile
+
+      - name: Compiling Code
+        run: yarn build
+
+      - name: Mocha Tests
+        run: yarn cover-local
+
+      - name: Performance Tests
+        run: sh test.sh
+
+      - name: Codecov
+        run: yarn codecov
+
+  test-s3:
+    name: Test S3
+    needs: test-local
+    runs-on: ubuntu-latest
+    env:
+      WINSTON_CONSOLE_LEVEL: none
+      EXPRESS_PORT: 1337
+      STORAGE_REPO: s3
+      SUB_FOLDER_ACTIVITIES: circleci/$CIRCLE_BUILD_NUM
+      SUB_FOLDER_AGENTS: SUB_FOLDER_AGENTS=circleci/$CIRCLE_BUILD_NUM
+      SUB_FOLDER_STATE: circleci/$CIRCLE_BUILD_NUM
+      SUB_FOLDER_STATEMENTS: circleci/$CIRCLE_BUILD_NUM
```

Comment:
Can we rename these folders and keep them agnostic of the service?

I don't see how `CIRCLE_BUILD_NUM` is defined.

Reply:
That's a built in @christopher-learningpool - https://circleci.com/docs/env-vars

Must have been some issue with concurrency to require that, Github probably has something equivalent.